src/components/Modal/README.md

```diff
@@ -246,6 +282,91 @@ React.createElement(() => {
             </ModalFooter>
           </Modal>
         )}
+        {modalOpen === 5 && (
+          <Modal
+            allowPrimaryActionOnEnterKey={false}
+            closeButtonRef={modalCloseButtonRef}
+            onCancel={(e) => {
+                console.log('cancel', e);
+            }}
+            onClose={(e) => {
+                console.log('close', e);
+            }}
+            primaryButtonRef={modalPrimaryButtonRef}
+          >
+            <ModalHeader>
+              <ModalTitle>Add new user using native form</ModalTitle>
+              <ModalCloseButton onClick={() => setModalOpen(false)} />
+            </ModalHeader>
+            <ModalBody>
+              <ModalContent>
+                <p>
+                  This is an example of a native form inside a modal.
+                  The difference is that the dialog is not controlled by React UI,
+                  but using native <code>&lt;form&gt;</code> element.
+                  This is useful when you need to use native form features
+                  like validation, submission, etc.
+                </p>
+                <p>
+                  First, you need to set <code>allowPrimaryActionOnEnterKey</code>
+                  to <code>false</code> and remove <code>onClick</code> from the
+                  primary button. Then, you need to set <code>form</code> attribute
+                  on the primary button to the <code>id</code> of the form to
+                  connect it with the form.
+                </p>
+                <p>
+                  Although we do not encourage using this approach, it is still
```

Comment:
Why don't we encourage the native solution? Is there something wrong when combining it with React? I'd like to see the reason mentioned here.